modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/plugins/fragments/components/CollectionDisplay.js

```diff
@@ -14,11 +14,53 @@
 
 import ClayIcon from '@clayui/icon';
 import classNames from 'classnames';
-import React from 'react';
+import React, {useEffect} from 'react';
+import {useDrag} from 'react-dnd';
+import {getEmptyImage} from 'react-dnd-html5-backend';
 
+import {useSelectItem} from '../../../app/components/Controls';
+import {LAYOUT_DATA_ITEM_TYPES} from '../../../app/config/constants/layoutDataItemTypes';
+import {useDispatch, useSelector} from '../../../app/store/index';
+import addItem from '../../../app/thunks/addItem';
 import Collapse from '../../../common/components/Collapse';
 
 const CollectionDisplayCard = () => {
+	const dispatch = useDispatch();
+	const store = useSelector(state => state);
+	const selectItem = useSelectItem();
+
+	const [, drag, preview] = useDrag({
+		end(_item, monitor) {
+			const result = monitor.getDropResult();
+
+			if (!result) {
+				return;
+			}
+
+			const {parentId, position} = result;
+
+			if (parentId) {
+				dispatch(
+					addItem({
+						itemType: LAYOUT_DATA_ITEM_TYPES.collection,
+						parentItemId: parentId,
+						position,
+						selectItem,
+						store,
+					})
+				);
+			}
+		},
+		item: {
+			name: Liferay.Language.get('collection-display'),
+			type: LAYOUT_DATA_ITEM_TYPES.collection,
+		},
+	});
+
+	useEffect(() => {
+		preview(getEmptyImage(), {captureDraggingState: true});
+	}, [preview]);
```

Comment:
I think this drag stuff is repeated in 3 places right? Maybe is time to move that into a hook ? 😛

Reply:
Or an util, useUtil('preview')

Reply:
xDDD

Reply:
Me gusta mucho ese hook, que cargue fincheros de utils con el loader y te los devuelva